tests/testsuite/rust_version.rs

```diff
@@ -0,0 +1,143 @@
+//! Tests for targets with `rust-version`.
+
+use cargo_test_support::project;
+
+#[cargo_test]
+fn rust_version_satisfied() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["rust-version"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+            rust-version = "1.1.1"
+            [[bin]]
+            name = "foo"
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build").masquerade_as_nightly_cargo().run();
+    p.cargo("build --ignore-rust-version")
+        .masquerade_as_nightly_cargo()
+        .run();
+}
+
+#[cargo_test]
+fn rust_version_too_high() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["rust-version"]
+
+            [project]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+            rust-version = "1.9876.0"
+            [[bin]]
+            name = "foo"
+        "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo()
+        .with_status(101)
+        .with_stderr(
+            "\
+            error: package `foo` cannot be built because it requires rustc ^1.9876.0, while the currently active rustc version is [..]
+",
+        )
+        .run();
+    p.cargo("build --ignore-rust-version")
+        .masquerade_as_nightly_cargo()
+        .run();
+}
+
+#[cargo_test]
+fn rust_version_dependency_fails() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["rust-version"]
+
+            [package]
+            name = "foo"
+            version = "0.0.1"
+            authors = []
+            [dependencies.bar]
+            path = "../bar"
+            #[dependencies.baz]
+            #path = "../baz"
+        "#,
+        )
+        .file("src/main.rs", "fn main(){}")
+        .build();
+    let _bar = project()
+        .at("bar")
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["rust-version"]
+
+            [package]
+            name = "bar"
+            version = "0.0.1"
+            authors = []
+            rust-version = "1.2345.0"
+        "#,
+        )
+        .file("src/lib.rs", "fn other_stuff(){}")
+        .build();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo()
+        .with_status(101)
+        .with_stderr(
+            "\
+error: package `bar` cannot be built because it requires rustc ^1.2345.0, while the currently active rustc version is [..]
```

Comment:
Hm, the caret here might be confusing for some people, particularly those not very familiar with Cargo or semver requirements.  Perhaps if the message could say something like "... it requires rustc 1.2345.0 or newer, ...".

Reply:
Fixed. This required a bit of surgery, with the result that the manifest propagates the value as a string, which gets checked for correctness (including parsing as `VersionReq`) when the manifest gets ingested, but is parsed again when checking in cargo_compile, where I `unwrap()` the error (since this is checked earlier). Is that okay?

Reply:
Yea, I think so.  If it's a problem later, it can be changed.